spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/AcknowledgeablePubsubMessage.java

```diff
@@ -16,16 +16,17 @@
 
 package org.springframework.cloud.gcp.pubsub.support;
 
-import java.util.Collections;
-
+import com.google.cloud.pubsub.v1.stub.SubscriberStub;
+import com.google.pubsub.v1.AcknowledgeRequest;
+import com.google.pubsub.v1.ModifyAckDeadlineRequest;
 import com.google.pubsub.v1.PubsubMessage;
 
 /**
  * A {@link PubsubMessage} wrapper that allows it to be acknowledged.
  *
  * <p>To acknowledge {@link AcknowledgeablePubsubMessage} in bulk, using a
- * {@link PubSubAcknowledger} is recommended. To do that, a user must first extract the ack IDs and
- * group them by subscription name.
+ * {@link org.springframework.cloud.gcp.pubsub.core.subscriber.PubSubSubscriberOperations#ack(java.util.Collection,
+ * String)} is recommended. To do that, a user must first extract the ack IDs and group them by subscription name.
```

Comment:
Let's consider to make such a "hard" logic in the template! I mean your idea to proceed with the `AcknowledgeablePubsubMessage` -based API makes sense. and we can group them by the `subscriptionName` already there.